cnf-certification-test/performance/suite.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/results"
+	"github.com/test-network-function/cnf-certification-test/internal/crclient"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/scheduling"
+	"github.com/test-network-function/cnf-certification-test/pkg/testhelper"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+)
+
+const (
+	SharedCPUScheduling    = "SHARED_CPU_SCHEDULING"
+	ExclusiveCPUScheduling = "EXCLUSIVE_CPU_SCHEDULING"
+)
+
+// All actual test code belongs below here.  Utilities belong above.
+var _ = ginkgo.Describe(common.PerformanceTestKey, func() {
+	logrus.Debugf("Entering %s suite", common.PerformanceTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		env = provider.GetTestEnvironment()
+	})
+	ginkgo.ReportAfterEach(results.RecordResult)
+
+	testID, tags := identifiers.GetGinkgoTestIDAndLabels(identifiers.TestSharedCPUPoolSchedulingPolicy)
+	ginkgo.It(testID, ginkgo.Label(tags...), func() {
+		var nonGuaranteedPodContainers = env.GetNonGuaranteedPodContainers()
+		testhelper.SkipIfEmptyAll(ginkgo.Skip, nonGuaranteedPodContainers)
+		testSchedOtherPolicyInSharedCPUPool(&env, nonGuaranteedPodContainers)
+	})
+})
+
+func testSchedOtherPolicyInSharedCPUPool(env *provider.TestEnvironment,
+	nonGuaranteedPodContainers []*provider.Container) {
+	nonCompliantContainers := make(map[*provider.Container][]int)
+	var compliantContainers []*provider.Container
+
+	for _, testContainer := range nonGuaranteedPodContainers {
+		logrus.Debugf("Processing %v", testContainer)
+
+		// Get the pid namespace
+		pidNamespace, err := crclient.GetContainerPidNamespace(testContainer, env)
+		if err != nil {
+			logrus.Errorf("unable to get pid namespace due to: %v", err)
+		}
+
+		// Get the list of process ids running in the pid namespace
+		pids := crclient.GetPidsFromPidNamespace(pidNamespace, testContainer)
+
+		// Check for the specified priority for each processes running in that pid namespace
+		isCompliant := processPidsCPUScheduling(pids, testContainer, nonCompliantContainers, SharedCPUScheduling)
+		if isCompliant {
+			compliantContainers = append(compliantContainers, testContainer)
+		}
+		logrus.Debugf("Processed %v", testContainer)
+	}
+	// Add success results to claim
+	if len(nonCompliantContainers) != 0 {
+		testhelper.AddTestResultLog("Non-compliant", nonCompliantContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
+	}
+	tnf.ClaimFilePrintf("Compliant containers", compliantContainers)
+}
+
+func processPidsCPUScheduling(pids []int, testContainer *provider.Container, nonCompliantContainers map[*provider.Container][]int, check string) (hasCPUSchedulingConditionSuccess bool) {
```

Comment:
Could `processPidsCPUScheduling` and `getProcessCPUScheduling` possibly be moved to another file within the new `performance` suite folder structure?  Maybe `performance/scheduling.go`?  I see there's already a `pkg/scheduling.go` so maybe that would be a good place?  Something like that.  In the other suites we have separated out the helper/logic funcs from the `suite.go` code just for organization.

Also would it be possible to build some unit tests around these funcs?

Reply:
Adding some unit tests.